test/runtest.py

```diff
@@ -1205,6 +1205,65 @@ def test_multi_threading_in_worker():
         # test multi-threading in the worker
         ray.get(test_multi_threading_in_worker.remote())
 
+    @unittest.skipIf(
+        os.environ.get("RAY_USE_XRAY") != "1",
+        "This test only works with xray.")
+    def testFreeObjectsMultiNode(self):
+        ray.worker._init(
+            start_ray_local=True,
+            num_local_schedulers=3,
+            num_workers=1,
+            num_cpus=[100, 5, 10],
```

Comment:
Let's do this with custom resources instead of CPUs/GPUs. That way we won't accidentally create tons and tons of workers.